### PR TITLE
chore: gpu base docker image

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: benchmark/camelyon/workflows.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def substrafl_fed_avg(
         local_code=[base / "common", base / "weldon_fedavg.py"],
         # Keeping editable_mode=True to ensure nightly test benchmarks are ran against main substrafl git ref
         editable_mode=True,
+        use_gpu=True,
```
Comment (thbcmlowk) [inline benchmark/camelyon/workflows.py:110]:
```suggestion
        use_gpu=False,
```

File: tests/remote/register/test_register.py [hunk 2]
```diff
@@ -43,8 +43,7 @@ def test_check_python_version_valid(version):
     register._check_python_version(version)
 
 
-def test_create_dockerfile(tmp_path, mocker, local_installable_module):
-    mocker.patch("substrafl.remote.register.register._get_base_docker_image", return_value="substratools-mocked")
+def test_create_cpu_dockerfile(tmp_path, mocker, local_installable_module):
```
Comment (SdgJlbl) [inline tests/remote/register/test_register.py:46]: Reminder for self: create the same with `use_gpu=True`

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: substrafl/remote/register/register.py [hunk 3]
```diff
@@ -29,14 +29,30 @@
...
+_CPU_BASE_IMAGE = """
+FROM python:{python_version}-slim
 
-DOCKERFILE_TEMPLATE = """
-FROM {docker_image}
+# update image
+RUN apt-get update -y
+"""
+
+_GPU_BASE_IMAGE = """
```
Comment (thbcmlowk) [inline substrafl/remote/register/register.py:39]: Do we want to do something about non-root users in GPU-friendly containers?

File: tests/remote/register/test_register.py [hunk 4]
```diff
@@ -37,14 +38,42 @@ def test_check_python_version(version):
...
+
+RUN apt-get install -y software-properties-common
+RUN add-apt-repository -y ppa:deadsnakes/ppa
+RUN apt-get -y upgrade
+
+# install python
+RUN apt-get install -y python3.11 python3.11-venv python3-pip
+
+"""
+    assert expected_dockerfile == _get_base_docker_image("3.11", use_gpu=True)
```
Comment (thbcmlowk) [inline tests/remote/register/test_register.py:73]: 🙏

**Review by thbcmlowk — APPROVED** [review]
Thank you for picking this up!

File: substrafl/remote/register/register.py (hunk 3 above)
Comment (SdgJlbl) [reply substrafl/remote/register/register.py:39]: The user management part has been factorized in the `DOCKERFILE_TEMPLATE` (lines 56 to 61).

**Comment by SdgJlbl** [comment]
Running MNIST on dev "worked" (there is an execution error `RuntimeError: Expected either a CPUGeneratorImplStateLegacy of size 5048 or a CPUGeneratorImplState of size 5056 but found the input RNG state size to be 16` that was already noted before, but building and running non torch code is working).

File: substrafl/remote/register/register.py [hunk 5]
```diff
@@ -29,21 +29,39 @@
...
+FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04
 
 # update image
-RUN apt update -y
+RUN apt-get update -y
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get install -y software-properties-common
```
Comment (guilhem-barthes) [inline substrafl/remote/register/register.py:46]: I would recommend do that in one layer and also remove the cache (cf https://docs.docker.com/build/building/best-practices/#apt-get)

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Pull request merged**